build/FrameworkPackage/CreateVpackFromNupkg.ps1

```diff
@@ -0,0 +1,90 @@
+<#
+.SYNOPSIS
+Use this script to create a VPack that can be pushed to the OS.
+The input to this script should be a non-prerelease .nupkg containing the MUX Framework Packages.
+This script does not push the vpack, but gives the required command to do so.
+#>
+[CmdLetBinding()]
+Param(
+    [string]$NugetPackageInputPath,
+    [string]$VPackDirectoryOutputPath
+    )
+
+$nupkgFileNameWithoutExtension = (Get-Item $NugetPackageInputPath).Basename
+
+if(!$nupkgFileNameWithoutExtension.StartsWith("Microsoft.UI.Xaml.","CurrentCultureIgnoreCase"))
+{
+    Write-Error "Expected input nupkg file to start with 'Microsoft.UI.Xaml'"
+    exit 1
+}
+
+$ver = $nupkgFileNameWithoutExtension -replace "Microsoft.UI.Xaml."
+$verParts = $ver.Split(".")
+if($verParts.Count -ne 3)
+{
+    Write-Error "Expected input nupkg file to have 3-place version number"
+    exit 1
+}
+$verMajor = $verParts[0]
+$verMinor = $verParts[1]
+$verPatch = $verParts[2]
+Write-Verbose "Version = '$verMajor.$verMinor.$verPatch'"
+
+Add-Type -AssemblyName System.IO.Compression.FileSystem
+$archive = [System.IO.Compression.ZipFile]::OpenRead($NugetPackageInputPath)
+
+function New-TemporaryDirectory {
+    $parent = [System.IO.Path]::GetTempPath()
+    $name = [System.IO.Path]::GetRandomFileName()
+    New-Item -ItemType Directory -Path (Join-Path $parent $name)
+}
+
+$tempDir = New-TemporaryDirectory
+$nugetUnpacked = $tempDir.FullName
+Write-Verbose "Temp Nuget directory: $nugetUnpacked"
+
+[System.IO.Compression.ZipFileExtensions]::ExtractToDirectory($archive, $nugetUnpacked)
```

Comment:
PowerShell actually has the built-in command [Expand-Archive](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.archive/expand-archive?view=powershell-6) - would that work here?

This obviously works too, it's just easiest to stick to PowerShell commands where possible.

Reply:
I didn't know about that. Interesting. I tried it out, but it will not extract a .nupkg file:

>Expand-Archive : .nupkg is not a supported archive file format. .zip is the only supported archive file format.

We could probably work around that by renaming the .nupkg to .zip first, but it seems simpler just to continue using System.IO.Compression.

Reply:
Yeah, that's fine.  Worth a shot. :)